uni/lib/model/entities/exam.dart

```diff
@@ -28,106 +39,68 @@ var _types = {
 /// Manages a generic Exam.
 ///
 /// The information stored is:
-/// - The Exam `subject`
 /// - The `begin` and `end` times of the Exam
+/// - The Exam `subject`
 /// - A List with the `rooms` in which the Exam takes place
-/// - The Exam `day`, `weekDay` and `month`
 /// - The Exam `type`
+
 class Exam {
+  late final DateTime begin;
+  late final DateTime end;
   late final String subject;
-  late final String begin;
-  late final String end;
-  late final DateTime beginDateTime;
-  late final DateTime endDateTime;
   late final List<String> rooms;
-  late final String day;
   late final String examType;
-  late final String weekDay;
-  late final String month;
-  late final String year;
-  late final DateTime date;
-
-  Exam.secConstructor(this.subject, this.beginDateTime, this.endDateTime,
-      String rooms, this.examType, this.weekDay) {
-    begin =
-        '${formattedString(beginDateTime.hour)}:${formattedString(beginDateTime.minute)}';
-    end =
-        '${formattedString(endDateTime.hour)}:${formattedString(endDateTime.minute)}';
-    this.rooms = rooms.split(',');
-    day = formattedString(beginDateTime.day);
-    month = months.keys.firstWhere(
-        (k) => months[k] == formattedString(beginDateTime.month),
-        orElse: () => '');
-    year = beginDateTime.year.toString();
-    date = DateTime(beginDateTime.year, beginDateTime.month, beginDateTime.day);
-  }
 
-  Exam(String schedule, this.subject, String rooms, String date, this.examType,
-      this.weekDay) {
-    final scheduling = schedule.split('-');
-    final splitDate = date.split('-');
-    this.date = DateTime.parse(date);
-
-    endDateTime = DateTime(
-        this.date.year,
-        this.date.month,
-        this.date.day,
-        int.parse(scheduling[1].split(':')[0]),
-        int.parse(scheduling[1].split(':')[1]));
-
-    beginDateTime = DateTime(
-        this.date.year,
-        this.date.month,
-        this.date.day,
-        int.parse(scheduling[0].split(':')[0]),
-        int.parse(scheduling[0].split(':')[1]));
-
-    begin = scheduling[0];
-    end = scheduling[1];
+  Exam.secConstructor(
+      this.subject, this.begin, this.end, String rooms, this.examType) {
     this.rooms = rooms.split(',');
-    year = splitDate[0];
-    day = splitDate[2];
-
-    month = months.keys
-        .firstWhere((k) => months[k] == splitDate[1], orElse: () => '');
   }
 
+  Exam(this.begin, this.end, this.subject, this.rooms, this.examType);
+
   /// Converts this exam to a map.
   Map<String, dynamic> toMap() {
     return {
       'subject': subject,
-      'begin': begin,
-      'end': end,
+      'begin': beginTime(),
+      'end': endTime(),
       'rooms': rooms.join(','),
-      'day': day,
+      'day': begin.day,
       'examType': examType,
-      'weekDay': weekDay,
-      'month': month,
-      'year': year
+      'weekDay': getWeekDay(),
+      'month': getMonth(),
+      'year': begin.year.toString()
```

Comment:
Since all the data type are Strings, perhaps we could make 'day' a string as well. Or does it have implications else where?

Reply:
@srissirs is right! 

exam.toMap is used to insert data in the database. I don't know if the insertInDatabase() function converts the number into text for a TEXT column, but I will not suggest it!

Use begin.day.toString() and change the variable type from Map<String, dynamic> to Map<String, String>.

The app_exams_database.dart file can be simplified, for sure... But that is out of scope for this PR

Reply:
Well noted! I'll correct it